lib/facter/resolvers/windows/ffi/ffi.rb

```diff
@@ -24,26 +24,33 @@
 
 module FFI
   WIN32FALSE = 0
-  END_OF_WCHAR_STRING = "\0\0".encode('UTF-16LE')
+  END_OF_WCHAR_STRING = (+"\0\0").force_encoding(Encoding::UTF_16LE)
```

Comment:
```suggestion
  END_OF_WCHAR_STRING = "\0".encode(Encoding::UTF_16LE).freeze
```

Reply:
I'm not sure I understand why this is necessary, isn't that already implicitly frozen from `# frozen_string_literal: true` on line 1?